### PR TITLE
rest/channel_spec: fix test checking error messages

### PR DESCRIPTION
**Pull request by Zariel**
Replace error message checks with error code checks. Error messages are not guaranteed to be consistent across versions/systems but the error codes are.

### CODE REVIEW

**Review by lmars — APPROVED** [review]
LGTM

**Review by sacOO7 — APPROVED** [review]
lgtm

**Review by sacOO7 — CHANGES_REQUESTED** [review]
Some of the acceptance tests are failing, can you take a look @Zariel

**Comment by jamienewcomb** [comment]
@sacOO7 can you approve and merge this if you are happy please

**Comment by sacOO7** [comment]
@Zariel I assume changes are fixed on the sandbox and will not break at client side going forward right?

File: spec/acceptance/rest/channel_spec.rb [hunk 1]
```diff
@@ -280,7 +280,7 @@
 
             context 'with an invalid client_id in the message' do
               it 'succeeds in the client library but then fails when published to Ably' do
-                expect { channel.publish([name: 'event', client_id: 'invalid']) }.to raise_error Ably::Exceptions::InvalidRequest, /mismatched clientId/
+                expect { channel.publish([name: 'event', client_id: 'invalid']) }.to raise_error Ably::Exceptions::InvalidRequest, /40012/
```
Comment (sacOO7) [inline spec/acceptance/rest/channel_spec.rb:283]:
```suggestion
                expect { channel.publish([name: 'event', client_id: 'invalid']) }.to raise_error(Ably::Exceptions::InvalidRequest, /40012/)
```

**Review by sacOO7 — APPROVED** [review]
lgtm

**Pull request merged**